decidim-forms/app/commands/decidim/forms/answer_questionnaire.rb

```diff
@@ -22,7 +22,10 @@ def initialize(form, current_user, questionnaire)
       def call
         return broadcast(:invalid) if @form.invalid? || user_already_answered?
 
-        answer_questionnaire
+        with_events do
```

Comment:
I am usually against binding modules together like we do here ( surveys depends on forms, yet forms should not know about surveys. I am ok with survey module to know about forms :) ). 

The `with_events` method, will dispatch an ActiveSupport::Notification event named "decidim.forms.answer_questionnaire:after" that would allow us to subscribe from survey module. 

You could register the following code initializer form surveys module, and remove the call and method `send_confirmation_email` from this controller. 

```

      initializer "decidim_proposals.moderation_content" do
        config.to_prepare do
          ActiveSupport::Notifications.subscribe("decidim.forms.answer_questionnaire:after") do |_event_name, data|
            component = data[:resource].questionaire_for.component
            answers = Decidim::Forms::QuestionnaireUserAnswers.for(data[:resource])
            user_answers = answers.select { |a| a.first.session_token == session_token }

            if component.manifest_name == "surveys" && component.settings.send_confirmation_email && answers.present?
              Decidim::Surveys::SurveyConfirmationMailer.confirmation(current_user, questionnaire, user_answers).deliver_later
            end
          end
        end
      end
```

As testing, you could add to `decidim-forms/spec/commands/decidim/forms/answer_questionnaire_spec.rb` 

```
          it_behaves_like "fires an ActiveSupport::Notification event", "decidim.forms.answer_questionnaire:before"
          it_behaves_like "fires an ActiveSupport::Notification event", "decidim.forms.answer_questionnaire:after"
```

Additionally, you can add more tests in the `decidim-surveys/spec/system`  to make sure the event is subscribed and dispatches email job.